data/test/scenarios/wml_tests/ScenarioWML/EventWML/events-test_filter_ability.cfg

```diff
@@ -296,3 +296,134 @@
 )}
 
 #undef FILTER_ABILITY_TEST
+
+#define FILTER_ABILITY_TEST_WITHOUT_VALUE
+    [event]
+        name=start
+        # Make sure the attacks hit
+        {FORCE_CHANCE_TO_HIT (id=bob) (id=alice) 100 ()}
+        {FORCE_CHANCE_TO_HIT (id=alice) (id=bob) 100 ()}
+        [modify_unit]
+            [filter]
+            [/filter]
+            # Make sure they don't die during the attacks
+            [status]
+                invulnerable=yes
+            [/status]
+        [/modify_unit]
+        [object]
+            silent=yes
+            [effect]
+                apply_to=new_ability
+                [abilities]
+                    [drains]
+                        [filter]
+                            [filter_location]
+                                time_of_day=neutral
+                            [/filter_location]
+                        [/filter]
+                    [/drains]
+                [/abilities]
+            [/effect]
+            [filter]
+                id=alice
+            [/filter]
+        [/object]
+        {VARIABLE triggers 0}
+    [/event]
+
+    [event]
+        name=side 1 turn 1
+        [do_command]
+            [move]
+                x=7,13
+                y=3,4
+            [/move]
+            [attack]
+                [source]
+                    x,y=13,4
+                [/source]
+                [destination]
+                    x,y=13,3
+                [/destination]
+            [/attack]
+        [/do_command]
+        [end_turn][/end_turn]
+    [/event]
+
+    [event]
+        name=side 2 turn
+        [do_command]
+            [attack]
+                [source]
+                    x,y=13,3
+                [/source]
+                [destination]
+                    x,y=13,4
+                [/destination]
+            [/attack]
+        [/do_command]
+        [end_turn][/end_turn]
+    [/event]
+#enddef
+
+#####
+# API(s) being tested: [event][filter][filter_ability]default_value
+##
+# Actions:
+# Use the common setup from FILTER_ABILITY_TEST_WITHOUT_VALUE.
+# Add an event with a filter matching Alice's drains ability.
+# Alice attacks Bob and then Bob attacks Alice, as defined in FILTER_ABILITY_TEST.
+##
+# Expected end state:
+# The filtered event is triggered exactly once.
```

Comment:
Needs `first_time_only=no` then.

Reply:
Huh? That doesn't seem correct if the intention is to assert that it's triggered exactly once…? Or am I misunderstanding something?

Reply:
It's needed to check that the filter works. With `first_time_only=yes` then it'll be triggered at most once even if the filter passes multiple times.

Reply:
@stevecotton first_time_only is never specified in the tests earlier in this file and that wasn't a problem, right? I copied them while only changing the interest parameters.

Reply:
I didn't look at any earlier tests, but I think that in general you should not use `first_time_only=yes` to ensure something in a test only happens once. It should use `first_time_only=no` and check that it occurs the expected number of times.